cmd/flux/cmd/test.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/edit"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/fluxinit"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/runtime"
+	"github.com/spf13/cobra"
+)
+
+// Test wraps the functionality of a single testcase statement,
+// to handle its execution and its pass/fail state.
+type Test struct {
+	ast *ast.Package
+	err error
+}
+
+// Create a new Test instance from an ast.Package.
```

Comment:
Nitpick: Tests should follow the godoc conventions as described [here](https://golang.org/doc/effective_go.html#commentary) so the first word of the godoc comment should be `NewTest`.

> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.
>
>     // Compile parses a regular expression and returns, if successful,
>     // a Regexp that can be used to match against text.
>     func Compile(str string) (*Regexp, error) {
>
> If every doc comment begins with the name of the item it describes, you > can use the doc subcommand of the go tool and run the output through grep. Imagine you couldn't remember the name "Compile" but were looking for the parsing function for regular expressions, so you ran the command,